airflow/providers/amazon/aws/operators/emr_add_steps.py

```diff
@@ -38,13 +38,14 @@ class EmrAddStepsOperator(BaseOperator):
     :type cluster_states: list
     :param aws_conn_id: aws connection to uses
     :type aws_conn_id: str
-    :param steps: boto3 style steps to be added to the jobflow. (templated)
-    :type steps: list
+    :param steps: boto3 style steps or reference to a steps file (must be '.json' or '.jinja2') to
+        be added to the jobflow. (templated)
+    :type steps: list|str
     :param do_xcom_push: if True, job_flow_id is pushed to XCom with key job_flow_id.
     :type do_xcom_push: bool
     """
     template_fields = ['job_flow_id', 'job_flow_name', 'cluster_states', 'steps']
-    template_ext = ()
+    template_ext = ('.json', '.jinja2')
```

Comment:
Nothing else uses `.jinja2` in Airflow, so I think we shouldn't create new pattern for just this operator.

Reply:
I can remove it. Though it's a widespread convention, and also helps distinguish between real file and template files.

Reply:
I now see that other operators with jsons such as `DruidOperator` and `SparkKubernetesOperator` don't use this. I'll update the PR to remove this.

Reply:
@ashb Removed `.jinja2`

Reply:
I think this is a good idea (separating file from templated files) it's just not something we do else where yet, so we should talk about making this a bigger change afterwards

Reply:
I think it's ready. Removed the `.jinja2` extension and also fixed a problem with the `template_ext` tuples.